src/App.tsx

```diff
@@ -1,25 +1,244 @@
+
+  const matchedCards = useMemo(
+    () => cards.filter((card) => card.isMatched),
+    [cards]
+  );
+
+  const shuffleArray = (array: CardType[]) => {
+    for (let i = array.length - 1; i > 0; i--) {
+      const j = Math.floor(Math.random() * (i + 1));
+      [array[i], array[j]] = [array[j], array[i]];
+    }
+    return array;
+  };
+
+  const handleCardClick = (clickedCard: CardType) => {
+    const updatedCards = [...cards];
+
+    const cardIndex = cards.findIndex((card) => card.id === clickedCard.id);
+
+    updatedCards[cardIndex] = {
+      ...updatedCards[cardIndex],
+      isFlipped: true,
+    };
+
+    setCards(updatedCards);
+
+    const flippedCards = updatedCards.filter(
+      (card) => card.isFlipped && !card.isMatched
+    );
+
+    if (flippedCards.length === 2) {
+      setMoves(moves + 1);
+      const [card1, card2] = flippedCards;
+      const matchedCards = updatedCards.map((card) => {
+        return {
+          ...card,
+          isMatched:
+            (card1.design === card2.design && card.design === card1.design) ||
+            card.isMatched,
+          isFlipped:
+            card1.design !== card2.design &&
+            ![card1.id, card2.id].includes(card.id) &&
+            card.isMatched,
+        };
+      });
+      setTimeout(() => {
+        setCards(matchedCards);
+      }, 1000);
+    } else {
+      setCards(updatedCards);
+    }
+  };
+  const handleDifficultyLevel = (level: string) => {
+    let numOfCards;
+    switch (level) {
+      case "easy":
+        numOfCards = 8;
+        break;
+      case "medium":
+        numOfCards = 12;
+        break;
+      case "hard":
+        numOfCards = 16;
+        break;
+      default:
+        numOfCards = 8;
+    }
+    setCards(shuffleArray(CARD_DATA.slice(0, numOfCards)));
+  };
+
+  const handleRestart = () => {
+    setCards([]);
+    setMoves(0);
+  };
+
+  const classes = useAppStyles();
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
+    <>
+      <div className={classes.appHeader}>
+        <AppBar position="static">
+          <Typography
+            textAlign="center"
+            variant="h4"
+            margin={2}
+            data-testid="title"
+          >
+            Match the Cards🃏
+          </Typography>
+        </AppBar>
+      </div>
+
+      {cards.length === matchedCards.length ? (
+        <>
+          {matchedCards.length > 0 ? (
+            <>
+              <Typography textAlign="center" variant="h4" m={2}>
+                Yay! You won. You took {moves} moves
+              </Typography>
+              <Grid display="flex" justifyContent="center" alignItems="center">
```

Comment:
For all grids, we don't need to give display flex. if we item it automatically considers it as display flex. Please change it everywhere.